### PR TITLE
Remove all `window.data`

### PR DESCRIPTION
**Pull request by Tin-Cheng**
remove all window.data, window.semester, window.terms
remove the use of ["val"]

### CODE REVIEW

File: client/src/components/CourseRow.js [hunk 1]
```diff
@@ -7,23 +7,25 @@ import removeCourse from '../timetable/removeCourse';
 
 import styles from './CourseRow.module.css';
 
+import DataContext from '../context';
 import infoIcon from '../images/info.png';
 import removeIcon from '../images/cross.png';
 
 const CourseRow = props => {
     // TODO: Remove usages of global variables
```
Comment (antonysiuming) [inline client/src/components/CourseRow.js:15]: You may remove this TODO now

**Review by antonysiuming — CHANGES_REQUESTED** [review]
Generally looks good to me, just a minor issue.

Please also merge the latest `main` branch to solve the conflict.

File: client/src/components/CourseRow.js (hunk 1 above)
Comment (Tin-Cheng) [reply client/src/components/CourseRow.js]: done

**Comment by Tin-Cheng** [comment]
conflicts resolved.

**Review by antonysiuming — APPROVED** [review]

**Pull request merged**